flake.nix

```diff
@@ -495,7 +563,7 @@
           #     builtins.attrValues self.project.${system}.hsPkgs
           #   )
           # );
-          hspkgs = builtins.map (x: self.project.${system}.hsPkgs.${x}.components.library) [
+          hspkgs = builtins.map (x: self.haddockProject.${system}.hsPkgs.${x}.components.library) [
```

Comment:
Isn't this change entirely unrelated? What does this do?

Reply:
OK, I see, it's for setting `-development` to false. You've put a lot of stuff into a single PR which makes it hard to merge.

Reply:
`haddockProject` is now used only for building haddocks (this rename makes that clear); the rest of the things use `flakeMatrix` (enumerates on 'development' flag as well as ghc version). All the Nix changes are all done to improve test coverage (across flag valus and compiler versions). I don't see any more Nix changes to be done in this branch.

I wanted 8.10 and development=true to be enabled in CI first, before basing other work on. Would you rather I open subsequent PRs (for porting existing tests) as separate PRs, instead of pushing to this branch?